### PR TITLE
enh(clib):Remove basic/interruption_error

### PR DESCRIPTION
**Pull request by sechkem**
## Description

Replacement of Macro basic_error  with the function msg_fmt 

REFS: MON-147186

### CODE REVIEW

File: engine/CMakeLists.txt
Comment (sechkem) [inline engine/CMakeLists.txt:1]: fix compilation

Comment (sechkem) [inline engine/CMakeLists.txt:1]: fix compilation

**Review by sechkem — COMMENTED** [review]
Remove Class basic and interruption and replace with msg_fmt

File: clib/src/logging/backend.cc [hunk 1]
```diff
@@ -170,19 +172,19 @@ void backend::show_thread_id(bool enable) {
...
-    buffer << "[" << timestamp::now().to_useconds() << "] ";
+    buffer = fmt::format("[{}] ", timestamp::now().to_useconds());
   if (_show_pid) {
-    buffer << "[" << getpid() << "] ";
+    buffer += fmt::format("[{}] ", getpid());
   }
   if (_show_thread_id)
-    buffer << "[" << std::this_thread::get_id() << "] ";
+    buffer += fmt::format("[{}] ", std::this_thread::get_id());
 }
```
Comment (bouda1) [inline clib/src/logging/backend.cc:188]: And then you update the string to return

File: clib/src/logging/engine.cc [hunk 2]
```diff
@@ -49,11 +50,13 @@ unsigned long engine::add(backend* obj,
...
-                           "failed: bad argument (null pointer)");
+    throw msg_fmt(
+        "add backend on the logging engine "
+        "failed: bad argument (null pointer)");
   if (verbose >= sizeof(unsigned int) * CHAR_BIT)
-    throw(basic_error() << "add backend on the logging engine "
-                           "failed: invalid verbose");
+    throw msg_fmt(
+        "add backend on the logging engine "
+        "failed: invalid verbose");
```
Comment (bouda1) [inline clib/src/logging/engine.cc:59]: strings to reformat

File: clib/src/logging/engine.cc [hunk 3]
```diff
@@ -125,8 +128,9 @@ bool engine::remove(unsigned long id) {
  */
 unsigned int engine::remove(backend* obj) {
   if (!obj)
-    throw(basic_error() << "remove backend on the logging engine "
-                           "failed:bad argument (null pointer)");
+    throw msg_fmt(
+        "remove backend on the logging engine "
+        "failed:bad argument (null pointer)");
```
Comment (bouda1) [inline clib/src/logging/engine.cc:133]: same here

File: clib/src/logging/file.cc [hunk 4]
```diff
@@ -119,24 +119,23 @@ void file::log(uint64_t types,
   (void)verbose;
   (void)size;
 
-  misc::stringifier header;
+  std::string header;
   _build_header(header);
```
Comment (bouda1) [inline clib/src/logging/file.cc:123]:
```suggestion
  std::string header = _build_header();
```

File: clib/src/logging/syslogger.cc [hunk 5]
```diff
@@ -78,11 +77,11 @@ void syslogger::log(uint64_t types,
   (void)verbose;
   (void)size;
 
-  misc::stringifier header;
+  std::string header;
   _build_header(header);
```
Comment (bouda1) [inline clib/src/logging/syslogger.cc:81]:
```suggestion
  std::string header = _build_header();
```

File: clib/src/process.cc [hunk 6]
```diff
@@ -500,33 +501,29 @@ pid_t process::_create_process_with_setpgid(char* const* args, char** env) {
   posix_spawnattr_t attr;
   int ret = posix_spawnattr_init(&attr);
   if (ret)
-    throw basic_error() << "cannot initialize spawn attributes: "
-                        << strerror(ret);
+    throw msg_fmt("cannot initialize spawn attributes: ", strerror(ret));
```
Comment (bouda1) [inline clib/src/process.cc:504]: missing {}

File: clib/src/process.cc [hunk 7]
```diff
@@ -500,33 +501,29 @@ pid_t process::_create_process_with_setpgid(char* const* args, char** env) {
...
   ret = posix_spawnattr_setflags(&attr, POSIX_SPAWN_SETPGROUP);
   if (ret) {
     posix_spawnattr_destroy(&attr);
-    throw basic_error() << "cannot set spawn flag: " << strerror(ret);
+    throw msg_fmt("cannot set spawn flag: ", strerror(ret));
```
Comment (bouda1) [inline clib/src/process.cc:508]: missing {}

File: clib/src/process.cc [hunk 8]
```diff
@@ -500,33 +501,29 @@ pid_t process::_create_process_with_setpgid(char* const* args, char** env) {
...
   }
   ret = posix_spawnattr_setpgroup(&attr, 0);
   if (ret) {
     posix_spawnattr_destroy(&attr);
-    throw basic_error()
-        << "cannot set process group ID of to-be-spawned process: "
-        << strerror(ret);
+    throw msg_fmt("cannot set process group ID of to-be-spawned process: ",
+                  strerror(ret));
```
Comment (bouda1) [inline clib/src/process.cc:514]: same here

File: clib/test/backend_test.hh [hunk 9]
```diff
@@ -49,9 +49,9 @@ class backend_test : public backend {
     (void)types;
     (void)verbose;
 
-    misc::stringifier header;
+    std::string header;
     _build_header(header);
-    _buffer.append(header.data(), header.size());
+    _buffer.append(header);
```
Comment (bouda1) [inline clib/test/backend_test.hh:54]:
```
  _buffer = _build_header();
```

File: connectors/perl/src/embedded_perl.cc [hunk 10]
```diff
@@ -96,8 +96,9 @@ pid_t embedded_perl::run(std::string const& cmd,
                          const shared_io_context& io_context) {
   // Check arguments.
   if (!fds)
-    throw basic_error() << "cannot run Perl script without "
-                           "fetching process' descriptors";
+    throw exceptions::msg_fmt(
+        "cannot run Perl script without "
+        "fetching process' descriptors");
```
Comment (bouda1) [inline connectors/perl/src/embedded_perl.cc:101]: string to reformat

File: connectors/perl/src/orders/parser.cc [hunk 11]
```diff
@@ -54,27 +55,30 @@ void parser::execute(const std::string& cmd) {
   char* ptr(nullptr);
   unsigned long long cmd_id(strtoull(cmd.c_str(), &ptr, 10));
   if (!cmd_id || *ptr)
-    throw basic_error() << "invalid execution request received:"
-                           " bad command ID ("
-                        << cmd.c_str() << ")";
+    throw msg_fmt(
+        "invalid execution request received:"
+        " bad command ID ({})",
+        cmd.c_str());
```
Comment (bouda1) [inline connectors/perl/src/orders/parser.cc:61]: string badly formatted and cmd.c_str() can be replaced by cmd

File: connectors/perl/src/orders/parser.cc [hunk 12]
```diff
@@ -54,27 +55,30 @@ void parser::execute(const std::string& cmd) {
...
   time_t timeout = static_cast<time_t>(strtoull(cmd.c_str() + pos, &ptr, 10));
   time_point ts_timeout = system_clock::now() + std::chrono::seconds(timeout);
 
   if (*ptr)
-    throw basic_error() << "invalid execution request received:"
-                           " bad timeout ("
-                        << cmd.c_str() + pos << ")";
+    throw msg_fmt(
+        "invalid execution request received:"
+        " bad timeout ({})",
```
Comment (bouda1) [inline connectors/perl/src/orders/parser.cc:71]: string to reformat

File: connectors/perl/src/orders/parser.cc [hunk 13]
```diff
@@ -54,27 +55,30 @@ void parser::execute(const std::string& cmd) {
...
   // Find start time.
   end = cmd.find('\0', pos);
   strtoull(cmd.c_str() + pos, &ptr, 10);
   if (*ptr)
-    throw basic_error() << "invalid execution request received:"
-                           " bad start time ("
-                        << cmd.c_str() + pos << ")";
+    throw msg_fmt(
+        "invalid execution request received:"
+        " bad start time ({})",
```
Comment (bouda1) [inline connectors/perl/src/orders/parser.cc:80]: same

File: connectors/ssh/src/main.cc [hunk 14]
```diff
@@ -77,13 +77,12 @@ int main(int argc, char* argv[]) {
...
-        throw basic_error() << "libssh2 initialization failed";
+        throw exceptions::msg_fmt("libssh2 initialization failed");
       {
         char const* version(libssh2_version(LIBSSH2_VERSION_NUM));
         if (!version)
-          throw basic_error()
-              << "libssh2 version is too old (>= " << LIBSSH2_VERSION
-              << " required)";
+          throw exceptions::msg_fmt(
+              "libssh2 version is too old (>= {} required)", LIBSSH2_VERSION);
```
Comment (bouda1) [inline connectors/ssh/src/main.cc:85]:
```suggestion
              "libssh2 version is too old (>= " LIBSSH2_VERSION " required)");
```

File: connectors/ssh/src/orders/parser.cc [hunk 15]
```diff
@@ -59,51 +60,57 @@ void parser::execute(const std::string& cmd) {
   char* ptr(nullptr);
   unsigned long long cmd_id(strtoull(cmd.c_str(), &ptr, 10));
   if (!cmd_id || *ptr)
-    throw basic_error() << "invalid execution request received:"
-                           " bad command ID ("
-                        << cmd << ")";
+    throw msg_fmt(
+        "invalid execution request received:"
+        " bad command ID ({})",
```
Comment (bouda1) [inline connectors/ssh/src/orders/parser.cc:65]: reformat

File: connectors/ssh/src/orders/parser.cc [hunk 16]
```diff
@@ -59,51 +60,57 @@ void parser::execute(const std::string& cmd) {
...
   end = cmd.find('\0', pos);
   time_t timeout(static_cast<time_t>(strtoull(cmd.c_str() + pos, &ptr, 10)));
 
   if (*ptr)
-    throw basic_error() << "invalid execution request received:"
-                           " bad timeout ("
-                        << cmd.c_str() + pos << ")";
+    throw msg_fmt(
+        "invalid execution request received:"
+        " bad timeout ({})",
```
Comment (bouda1) [inline connectors/ssh/src/orders/parser.cc:75]: reformat

File: connectors/ssh/src/orders/parser.cc [hunk 17]
```diff
@@ -59,51 +60,57 @@ void parser::execute(const std::string& cmd) {
...
   // Find start time.
   end = cmd.find('\0', pos);
   time_t start_time(static_cast<time_t>(strtoull(cmd.c_str() + pos, &ptr, 10)));
   if (*ptr || !start_time)
-    throw basic_error() << "invalid execution request received:"
-                           " bad start time ("
-                        << cmd.c_str() + pos << ")";
+    throw msg_fmt(
+        "invalid execution request received:"
+        " bad start time ({})",
```
Comment (bouda1) [inline connectors/ssh/src/orders/parser.cc:85]: reformat

File: connectors/ssh/src/orders/parser.cc [hunk 18]
```diff
@@ -59,51 +60,57 @@ void parser::execute(const std::string& cmd) {
...
   // Find command to execute.
   end = cmd.find('\0', pos);
   std::string cmdline(cmd.substr(pos, end - pos));
   if (cmdline.empty())
-    throw basic_error() << "invalid execution request received:"
-                           " bad command line ("
-                        << cmd.c_str() + pos << ")";
+    throw msg_fmt(
+        "invalid execution request received:"
+        " bad command line ({})",
```
Comment (bouda1) [inline connectors/ssh/src/orders/parser.cc:94]: reformat

File: connectors/ssh/src/orders/parser.cc [hunk 19]
```diff
@@ -59,51 +60,57 @@ void parser::execute(const std::string& cmd) {
...
   com::centreon::connector::orders::options::pointer opt(
       std::make_shared<com::centreon::connector::orders::options>());
   try {
     opt->parse(cmdline);
     if (opt->get_commands().empty())
-      throw basic_error() << "invalid execution request "
-                             "received: bad command line ("
-                          << cmd.c_str() + pos << ")";
+      throw msg_fmt(
+          "invalid execution request "
```
Comment (bouda1) [inline connectors/ssh/src/orders/parser.cc:102]: reformat

File: connectors/ssh/src/orders/parser.cc [hunk 20]
```diff
@@ -59,51 +60,57 @@ void parser::execute(const std::string& cmd) {
...
     if (opt->get_timeout() &&
         opt->get_timeout() < static_cast<unsigned int>(timeout))
       ts_timeout =
           system_clock::now() + std::chrono::seconds(opt->get_timeout());
     else if (opt->get_timeout() > static_cast<unsigned int>(timeout))
-      throw basic_error() << "invalid execution request "
-                             "received: timeout > to monitoring engine timeout";
+      throw msg_fmt(
+          "invalid execution request "
+          "received: timeout > to monitoring engine timeout");
```
Comment (bouda1) [inline connectors/ssh/src/orders/parser.cc:113]: reformat

File: connectors/ssh/src/sessions/session.cc [hunk 21]
```diff
@@ -672,17 +673,17 @@ void session::handshake_handler(int retval,
...
-        e << "host fingerprint mismatch with known_hosts file "
-          << known_hosts_file;
+        e = fmt::format("host fingerprint mismatch with known_hosts file {}",
+                        known_hosts_file);
       else
-        e << "unknown error";
-      throw e;
+        e = "unknown error";
+      throw msg_fmt("host '{}' is not known or could not be validated: {}",
+                    _creds.get_host(), e);
```
Comment (bouda1) [inline connectors/ssh/src/sessions/session.cc:686]: In the three cases, we throw an exception.
I propose to throw it directly in the three cases, so no need of e

File: engine/tests/commands/bin_connector_test_run.cc [hunk 22]
```diff
@@ -119,17 +118,19 @@ static void query_execute(char const* q) {
...
+        "invalid query execute: "
+        "invalid command_id");
   startptr = endptr + 1;
   uint32_t timeout(strtol(startptr, &endptr, 10));
   if (startptr == endptr)
-    throw basic_error() << "invalid query execute: "
-                           "invalid is_executed";
+    throw msg_fmt(
+        "invalid query execute: "
+        "invalid is_executed");
```
Comment (bouda1) [inline engine/tests/commands/bin_connector_test_run.cc:129]: strings to reformat

**Review by bouda1 — CHANGES_REQUESTED** [review]

File: clib/inc/com/centreon/logging/backend.hh [hunk 23]
```diff
@@ -20,13 +20,10 @@
...
 #include <mutex>
+#include <string>
 
 namespace com::centreon {
 
-namespace misc {
-class stringifier;
-}
-
 namespace logging {
```
[the remark this answers is not in the document]
Comment (sechkem) [reply clib/inc/com/centreon/logging/backend.hh:27]: yes, will fix

File: clib/src/handle_manager.cc [hunk 24]
```diff
@@ -76,8 +77,9 @@ void handle_manager::add(handle* h, handle_listener* hl, bool is_threadable) {
     _handles.insert(item);
     _recreate_array = true;
   } else
-    throw basic_error() << "attempt to add handle "
-                           "already monitored by handle manager";
+    throw exceptions::msg_fmt(
+        "attempt to add handle "
+        "already monitored by handle manager");
```
[the remark this answers is not in the document]
Comment (sechkem) [reply clib/src/handle_manager.cc:82]: will fix

File: clib/src/handle_manager.cc [hunk 25]
```diff
@@ -59,15 +59,16 @@ handle_manager::~handle_manager() noexcept {
...
+    throw exceptions::msg_fmt("attempt to add null listener in handle manager");
 
   // Check native handle.
   native_handle nh(h->get_native_handle());
   if (nh == native_handle_null)
-    throw basic_error() << "attempt to add handle with invalid native "
-                           "handle in the handle manager";
+    throw exceptions::msg_fmt(
+        "attempt to add handle with invalid native "
+        "handle in the handle manager");
```
[the remark this answers is not in the document]
Comment (sechkem) [reply clib/src/handle_manager.cc:71]: will fix

File: clib/src/io/file_entry.cc [hunk 26]
```diff
@@ -188,7 +189,7 @@ void file_entry::refresh() {
     memset(&_sbuf, 0, sizeof(_sbuf));
   else if (stat(_path.c_str(), &_sbuf)) {
     char const* msg(strerror(errno));
-    throw(basic_error() << "get file information failed: " << msg);
+    throw msg_fmt("get file information failed: ", msg);
```
[the remark this answers is not in the document]
Comment (sechkem) [reply clib/src/io/file_entry.cc:192]: will fix

File: clib/src/io/file_stream.cc [hunk 27]
```diff
@@ -126,9 +127,10 @@ com::centreon::native_handle file_stream::get_native_handle() {
     retval = fileno(_stream);
     if (retval < 0) {
       char const* msg(strerror(errno));
-      throw(basic_error() << "could not get native handle from "
-                             "file stream: "
-                          << msg);
+      throw msg_fmt(
+          "could not get native handle from "
+          "file stream: {}",
```
[the remark this answers is not in the document]
Comment (sechkem) [reply clib/src/io/file_stream.cc:132]: will fix

File: clib/src/logging/backend.cc [hunk 28]
```diff
@@ -170,19 +172,19 @@ void backend::show_thread_id(bool enable) {
  *
  *  @param[out] buffer  The buffer to fill.
  */
-void backend::_build_header(misc::stringifier& buffer) {
+void backend::_build_header(std::string& buffer) {
```
[the remark this answers is not in the document]
Comment (sechkem) [reply clib/src/logging/backend.cc:175]: will fix

File: clib/src/logging/backend.cc (hunk 1 above)
Comment (sechkem) [reply clib/src/logging/backend.cc:188]: will fix

File: clib/src/logging/engine.cc (hunk 2 above)
Comment (sechkem) [reply clib/src/logging/engine.cc:59]: will fix

File: broker/test/CMakeLists.txt [hunk 29]
```diff
@@ -127,7 +127,7 @@ add_executable(
   ${TESTS_DIR}/processing/feeder.cc
   ${TESTS_DIR}/time/timerange.cc
   ${TESTS_DIR}/rpc/brokerrpc.cc
-  ${TESTS_DIR}/exceptions.cc
```
[the remark this answers is not in the document]
Comment (sechkem) [reply broker/test/CMakeLists.txt]: Discard the change , need for broker/core/test

File: broker/neb/inc/com/centreon/logging/backend.hh [hunk 30]
```diff
@@ -19,13 +19,11 @@
 #ifndef CC_LOGGING_BACKEND_HH
 #define CC_LOGGING_BACKEND_HH
```
[the remark this answers is not in the document]
Comment (sechkem) [reply broker/neb/inc/com/centreon/logging/backend.hh]: will fix

File: connectors/ssh/src/sessions/session.cc (hunk 21 above)
Comment (sechkem) [reply connectors/ssh/src/sessions/session.cc]: will fix

File: engine/tests/commands/bin_connector_test_run.cc (hunk 22 above)
Comment (sechkem) [reply engine/tests/commands/bin_connector_test_run.cc]: will fix

File: connectors/ssh/src/orders/parser.cc (hunk 20 above)
Comment (sechkem) [reply connectors/ssh/src/orders/parser.cc]: will fix

File: connectors/ssh/src/orders/parser.cc (hunk 18 above)
Comment (sechkem) [reply connectors/ssh/src/orders/parser.cc]: will fix

File: clib/src/handle_manager.cc (hunk 24 above)
[the remark this answers is not in the document]
Comment (sechkem) [reply clib/src/handle_manager.cc]: will fix

File: clib/src/logging/engine.cc (hunk 3 above)
Comment (sechkem) [reply clib/src/logging/engine.cc]: will fix

File: clib/src/logging/file.cc (hunk 4 above)
Comment (sechkem) [reply clib/src/logging/file.cc]: will fix

File: clib/src/logging/syslogger.cc (hunk 5 above)
Comment (sechkem) [reply clib/src/logging/syslogger.cc]: will fix

File: clib/src/process.cc (hunk 6 above)
Comment (sechkem) [reply clib/src/process.cc]: will fix

File: clib/src/process.cc (hunk 7 above)
Comment (sechkem) [reply clib/src/process.cc]: will fix

File: clib/src/process.cc (hunk 8 above)
Comment (sechkem) [reply clib/src/process.cc]: will fix

File: clib/test/backend_test.hh (hunk 9 above)
Comment (sechkem) [reply clib/test/backend_test.hh]: will fix

File: connectors/perl/src/orders/parser.cc (hunk 11 above)
Comment (sechkem) [reply connectors/perl/src/orders/parser.cc]: will fix

File: connectors/perl/src/orders/parser.cc (hunk 12 above)
Comment (sechkem) [reply connectors/perl/src/orders/parser.cc]: will fix

File: connectors/perl/src/orders/parser.cc (hunk 13 above)
Comment (sechkem) [reply connectors/perl/src/orders/parser.cc]: will fix

File: connectors/ssh/src/main.cc (hunk 14 above)
Comment (sechkem) [reply connectors/ssh/src/main.cc]: will fix

File: connectors/ssh/src/orders/parser.cc (hunk 15 above)
Comment (sechkem) [reply connectors/ssh/src/orders/parser.cc]: will fix

File: connectors/ssh/src/orders/parser.cc (hunk 16 above)
Comment (sechkem) [reply connectors/ssh/src/orders/parser.cc]: will fix

File: connectors/ssh/src/orders/parser.cc (hunk 17 above)
Comment (sechkem) [reply connectors/ssh/src/orders/parser.cc]: will fix

File: connectors/perl/src/embedded_perl.cc (hunk 10 above)
Comment (sechkem) [reply connectors/perl/src/embedded_perl.cc]: will fix

**Review by bouda1 — APPROVED** [review]

**Review by bouda1 — APPROVED** [review]

**Pull request merged**